### DISCUSSION TITLE
how to get response cookies ?

### DISCUSSION THREAD

**Discussion by blogwy** [sindresorhus/got#1703] [Q&A] [answered]
#### What would you like to discuss?

...

#### Checklist

- [ ] I have read the documentation.

**Comment by spaledk** [comment]
I have the same problem. Tell me, has anyone found a way to get cookies from a request?
I tried the same way as @edis-v . That didn't work

**Reply by LeonAlvarez** [reply]
```ts
const {headers}  =  await got(url)

const cookies =  headers['set-cookie'].reduce((cookies: any, c: string) => {
  const [key, val] = c.split(';')[0].split('=');
  cookies[key] = val;

  return cookies;
}, {});

```

**Comment by sindresorhus** [comment, marked as answer]
You can access cookies through the `set-cookie` header in the response:

```javascript
const response = await got('https://example.com');
const cookies = response.headers['set-cookie'];

console.log(cookies); // Array of cookie strings
```

If you need to parse them into a more useable format, you can do something like:

```javascript
const {headers} = await got('https://example.com');
const cookieStrings = headers['set-cookie'] || [];

const cookies = cookieStrings.reduce((acc, cookie) => {
	const [keyValue] = cookie.split(';');
	const [key, value] = keyValue.split('=');
	acc[key] = value;
	return acc;
}, {});

console.log(cookies); // {cookieName: 'cookieValue', ...}
```

For more advanced cookie handling (like preserving cookies accross requests), you might want to use a cookie jar library like [`tough-cookie`](https://github.com/salesforce/tough-cookie), which Got supports through the `cookieJar` option.
